openstack/compute/v2/servers/results.go

```diff
@@ -92,6 +99,41 @@ func (r GetPasswordResult) ExtractPassword(privateKey *rsa.PrivateKey) (string,
 	return s.Password, err
 }
 
+func (r ShowConsoleOutputResult) ExtractConsoleOutput() (string, error) {
+	if r.Err != nil {
+		return "", r.Err
+	}
+
+	var s struct {
+		Output string `json:"output"`
+	}
+
+	err := r.ExtractInto(&s)
+	return s.Output, err
+}
+
+func GetHostKeyFromConsole(consoleOutput string) (string, error) {
+	start := `-----BEGIN SSH HOST KEY KEYS-----\r`
+	end := `-----END SSH HOST KEY KEYS-----\r`
+	searchFlag := false
+	lines := strings.Split(consoleOutput, "\n")
+	for _, value := range lines {
+		if value == start {
+			searchFlag = true
+		}
+
+		if value == end {
+			searchFlag = false
+		}
+
+		if searchFlag && strings.HasPrefix(value, "ssh-rsa") {
+			return value, nil
+		}
+	}
+
+	return "", errors.New("ssh host key not found")
+}
```

Comment:
I appreciate what this function is doing, but I think it's too niche to be included in Gophercloud as a whole. This function could easily be added to the application which needs to parse the console output. Let's remove this.